shortfin/python/shortfin_apps/llm/components/invocation.py

```diff
@@ -355,35 +335,67 @@ async def run(self):
             RuntimeError: No available entry point for given batch size.
         """
         try:
-            req_count = self.llm_task.req_count
+            req_count = self._llm_task.req_count
 
             # Select an entrypoint for the batch.
-            entrypoints = self.functions
+            entrypoints = self._functions
             for bs, fn in entrypoints.items():
                 if bs >= req_count:
                     break
             else:
                 raise RuntimeError(f"No available entry point for bs {req_count}")
 
-            args = await self.llm_task.prepare_args(bs)
+            args = await self._llm_task.prepare_args(bs)
             args_device = [arg.device for arg in args]
 
             # Invoke VMFB. Logits are of shape [bs, bsl, d].
             results = await fn(*args_device, fiber=self.fiber)
 
-            logits = results[0]
             indices = None
+            logits = results[0]
             if len(results) > 1:
                 indices = results[1]
 
-            logits, indices = await self.llm_task.process_results(
+            logits, indices = await self._llm_task.process_results(
                 args,
                 logits,
                 indices,
-                self.device0,
+                self._device0,
             )
 
-            self.responder.set_success(logits, indices)
+            self._responder.set_success(logits, indices)
 
         except Exception as exception:
-            self.responder.set_failure(exception)
+            self._responder.set_failure(exception)
+
+
+def build_invocation_process(
+    phase: InferencePhase,
+    exec_requests: List[LlmInferenceExecRequest],
+    fiber: sf.Fiber,
+    array_cache: DeviceArrayCache,
+    page_cache: BasePagedAttentionCache,
+    functions: dict[int, sf.ProgramFunction],
+    program_isolation: sf.ProgramIsolation,
+    responder: LlmTaskResponder,
+) -> LlmInvocationProcess:
+    task_inputs = LlmTaskInput.from_exec_requests(
+        exec_requests=exec_requests,
+        seq_stride=page_cache.tokens_per_page,
+    )
+
+    llm_task_cls = PrefillTask if phase == InferencePhase.PREFILL else DecodeTask
```

Comment:
This feels wrong, having an enum just for invoking the wrong contruction means you are probably making this decision lower in the stack than you should. If you create the `LlmTask` in `batcher` you won't have to worry about this selection logic.